front_end/src/app/(campaigns-registration)/bridgewater/components/registration-status.tsx

```diff
@@ -0,0 +1,260 @@
+                />
+              </div>
+            ),
+          });
+        }}
+      >
+        Register for the Contest
+      </BWRegisterButton>
+    </>
+  );
+
+  const EligibilityBox: FC<PropsWithChildren<{ isEligible: boolean }>> = ({
+    children,
+    isEligible,
+  }) => {
+    return (
+      <div
+        className={cn(
+          "flex items-center gap-3 text-nowrap rounded bg-olive-400 px-2 py-1 dark:bg-olive-400-dark",
+          !isEligible && "bg-salmon-400 dark:bg-salmon-400-dark"
+        )}
+      >
+        <FontAwesomeIcon icon={isEligible ? faCheckCircle : faXmarkCircle} />
+        <span className="font-normal text-gray-800 dark:text-gray-800-dark">
+          {children}
+        </span>
+        <span
+          className={cn(
+            "ml-auto font-bold text-olive-800 dark:text-olive-800-dark",
+            !isEligible && "text-salmon-800 dark:text-salmon-800-dark"
+          )}
+        >
+          {isEligible ? "Eligible *" : "Not Eligible *"}
+        </span>
+      </div>
+    );
+  };
+
+  const LoggedInAndRegisteredFragment: FC<{ eligibleBoth: boolean }> = ({
+    eligibleBoth,
+  }) => {
+    const { setCurrentModal } = useModal();
+
+    return (
+      <>
+        <HeadingText
+          heading={"You’re all set!"}
+          subheading={
+            "Your registration is confirmed. Here’s your leaderboard eligibility:"
+          }
+        />
+
+        <div className="flex w-full flex-col gap-1 text-base text-gray-800 dark:text-gray-800-dark">
+          <EligibilityBox isEligible={true}>Open Prize Pool</EligibilityBox>
+
+          <EligibilityBox isEligible={eligibleBoth}>
+            Undergrad Prize Pool
+          </EligibilityBox>
+        </div>
+
+        <p className="mb-0 mt-5 text-sm text-gray-0 dark:text-gray-0-dark xs:text-base sm:text-sm md:text-lg ">
+          Ready to get started? Try the{" "}
+          <span
+            className="cursor-pointer underline"
+            onClick={() => setCurrentModal({ type: "onboarding" })}
+          >
+            Forecasting Tutorial
+          </span>{" "}
+          or explore some{" "}
+          <Link href="/tournaments/bridgewater/">Warmup Questions</Link> to
+          sharpen your skills!
+        </p>
+
+        <div className="flex items-start gap-1.5 text-xs text-blue-400 dark:text-blue-400-dark">
+          <span className="my-0">*</span>
+          <p className="my-0 text-left">
+            Displayed prize-pool eligibility is based solely on your
+            undergraduate status. For full eligibility details see the{" "}
+            <Link href="contest-rules">contest rules</Link>.
+          </p>
+        </div>
+      </>
+    );
+  };
+
+  const campaigns = currentUser?.registered_campaigns.filter(
+    ({ key }) => key == CAMPAIGN_KEY
+  );
+
+  const registered = campaigns && campaigns.length > 0;
+  const campaignDetails =
+    registered && (campaigns[0]?.details as { undergrad: boolean });
```

Comment:
Is there a strong correlation between campaign key value and `details` type? If so, we better to apply proper typing for the `details`

Reply:
Yes and no. The types different from campaign to campaign, and also the data is different depending on whether the user checks a box or not.

We could create a type, ofc, but I think in this case it would be an unnecessary overhead - it feels to me we would be working for tsc instead of tsc working for us.